s3connector/s3.go

```diff
@@ -0,0 +1,133 @@
+package s3connector
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3Config struct {
+	Endpoint    string `json:"endpoint"`
+	Region      string `json:"region"`
+	Bucket      string `json:"bucket"`
+	AccessKey   string `json:"accessKey"`
+	SecretKey   string `json:"secretKey"`
+	Prefix      string `json:"prefix"`
+	StorageType string `json:"storageType"`
+}
+
+type S3ObjectRange struct {
+	Start int64 `json:"start"`
+	End   int64 `json:"end"`
+}
+
+type S3ObjectPath struct {
+	Bucket string         `json:"bucket"`
+	Key    string         `json:"key"`
+	Range  *S3ObjectRange `json:"range,omitempty"`
+}
+
+type ObjectStorage interface {
+	StoreObject(key string, value io.ReadSeeker) (S3ObjectPath, error)
+	DeleteObject(key string) error
+	GetObject(objPath S3ObjectPath) (io.ReadCloser, error)
+	GetBucket() string
+}
+
+type s3ObjectStorage struct {
+	ObjectStorage
+	session      *session.Session
+	bucket       string
+	storageClass string
+	prefix       string
+}
+
+func NewS3ObjectStorage(config S3Config) (ObjectStorage, error) {
+	awsConf := &aws.Config{
+		Region:           aws.String(config.Region),
+		S3ForcePathStyle: aws.Bool(true),
+	}
+	if config.Endpoint != "" {
+		awsConf.Endpoint = aws.String(config.Endpoint)
+	}
+	if config.AccessKey != "" && config.SecretKey != "" {
+		awsConf.Credentials = credentials.NewStaticCredentials(config.AccessKey, config.SecretKey, "")
+	}
+	session, err := session.NewSession(awsConf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new AWS session: %w", err)
+	}
+
+	s3ObjectStorageInstance := &s3ObjectStorage{session: session, bucket: config.Bucket, storageClass: config.StorageType, prefix: config.Prefix}
+
+	// Check if the bucket exists
+	if !s3ObjectStorageInstance.BucketExists(config.Bucket) {
+		return nil, fmt.Errorf("bucket %s does not exist", config.Bucket)
+	}
+	return s3ObjectStorageInstance, nil
+}
+
+func (s *s3ObjectStorage) GetBucket() string {
+	return s.bucket
+}
+
+func (s *s3ObjectStorage) BucketExists(bucket string) bool {
+	_, err := s3.New(s.session).HeadBucket(&s3.HeadBucketInput{Bucket: aws.String(bucket)})
+	return err == nil
+}
+
+func (s *s3ObjectStorage) StoreObject(key string, value io.ReadSeeker) (S3ObjectPath, error) {
```

Comment:
I am not sure. 
Is it make sense to have "objPath S3ObjectPath" instead of "key string" here as we have in the get?